eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -903,7 +889,7 @@ private function internalLoadContent(array $IdVersionTranslationPairs): array
                 'c',
                 'ezcontentobject_version',
                 'v',
-                'c.id = v.contentobject_id'
+                'c.id = v.contentobject_id AND v.version = ' . ($version ?: 'c.current_version')
```

Comment:
Please use `\Doctrine\DBAL\Query\Expression\ExpressionBuilder` instead of creating SQL using string concatenation 😉